nuxeo-services/nuxeo-platform-filemanager-core/src/test/java/org/nuxeo/ecm/platform/filemanager/TestCSVZipImporter.java

```diff
@@ -0,0 +1,197 @@
+ *     Nuxeo - initial API and implementation
+ *     Jackie Aldama <jaldama@nuxeo.com>
+ * $Id$
+ */
+package org.nuxeo.ecm.platform.filemanager;
+
+import java.io.File;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.zip.ZipFile;
+
+import javax.inject.Inject;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.common.utils.FileUtils;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.Blobs;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.PathRef;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.ecm.core.test.annotations.Granularity;
+import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.ecm.platform.filemanager.api.FileImporterContext;
+import org.nuxeo.ecm.platform.filemanager.api.FileManager;
+import org.nuxeo.ecm.platform.filemanager.service.extension.CSVZipImporter;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertSame;
+
+@RunWith(FeaturesRunner.class)
+@Features(CoreFeature.class)
+@RepositoryConfig(init = RepositoryInit.class, cleanup = Granularity.METHOD)
+@Deploy("org.nuxeo.ecm.platform.types.api")
+@Deploy("org.nuxeo.ecm.platform.types.core")
+@Deploy("org.nuxeo.ecm.platform.filemanager.core")
+@Deploy("org.nuxeo.ecm.platform.filemanager.core.tests:faceted-tag-core-types-contrib.xml")
+public class TestCSVZipImporter {
+
+    @Inject
+    protected CoreSession coreSession;
+
+    protected DocumentModel workspace1;
+
+    protected DocumentModel workspace2;
+
+    protected DocumentModel wsRoot;
+
+    private static File getArchiveFile(String file) {
+        return new File(FileUtils.getResourcePathFromContext(file));
+    }
+
+    @Before
+    public void createTestDocuments() throws Exception {
+        wsRoot = coreSession.getDocument(new PathRef("/default-domain/workspaces"));
+
+        workspace1= coreSession.createDocumentModel(wsRoot.getPathAsString(), "ws1", "Workspace");
+        workspace1.setProperty("dublincore", "title", "test WS1");
+        workspace1 = coreSession.createDocument(workspace1);
+
+        workspace2= coreSession.createDocumentModel(wsRoot.getPathAsString(), "ws2", "Workspace");
+        workspace2.setProperty("dublincore", "title", "test WS2");
+        workspace2 = coreSession.createDocument(workspace2);
+    }
+
+    @Test
+    public void testArchiveDetection() throws Exception {
+        ZipFile archive = CSVZipImporter.getArchiveFileIfValid(getArchiveFile("test-data/testCSVArchive.zip"));
+        assertNotNull(archive);
+        archive.close();
+    }
```

Comment:
The method getArchiveFileIfValid returns null in several cases you need to try as well to make sure it behaves the way you want when.